src/Application.vala

```diff
@@ -237,15 +238,20 @@ public class Terminal.Application : Gtk.Application {
         if (options.lookup ("execute", "^a&ay", out commands)) {
             for (var i = 0; commands[i] != null; i++) {
                 if (commands[i] != "\0") {
-                    window.add_tab_with_working_directory (working_directory, commands[i], new_tab);
+                    window.add_tab_with_working_directory (working_directory, commands[i]);
                 }
             }
         } else if (options.lookup ("commandline", "^&ay", out command) && command != "\0") {
-            window.add_tab_with_working_directory (working_directory, command, new_tab);
-        } else {
-            window.add_tab_with_working_directory (working_directory, null, new_tab);
+            window.add_tab_with_working_directory (working_directory, command);
+        } else if (wd_was_requested) {
+            window.add_tab_with_working_directory (working_directory, null);
+        } else if (new_tab) {
+            window.add_tab_with_working_directory (working_directory, null); //FIXME Should we honor "follow-last-tab" setting?
         }
 
+        // Only if no other tabs, add default tab
+        add_default_tab (window, false);
+
```

Comment:
this won't respect the current working directory as the others.

Reply:
Whether or not it respects the current working directory depends on the "follow-last-tab" setting (that is what the *default* tab does) since it uses the action-new-tab. This is for consistency with other situations where there is no specified tab.

Reply:
this is already being inconsistent with the others command line options handling, where the current working directory is always respected. the code in the action-new-tab won't respect it either, it will either use the last tab directory or the current working directory of the first invocation.